src/StaticWebAssetsSdk/Tasks/Compression/ResolveCompressedAssets.cs

```diff
@@ -51,56 +53,9 @@ public override bool Execute()
             return true;
         }
 
-        // Scan the provided candidate assets and determine which ones have already been detected for compression and in which formats.
-        var existingCompressionFormatsByAssetItemSpec = new Dictionary<string, HashSet<string>>();
-        foreach (var asset in CandidateAssets)
-        {
-            if (IsCompressedAsset(asset))
-            {
-                var relatedAssetItemSpec = asset.GetMetadata("RelatedAsset");
-
-                if (string.IsNullOrEmpty(relatedAssetItemSpec))
-                {
-                    Log.LogError(
-                        "The asset '{0}' was detected as compressed but didn't specify a related asset.",
-                        asset.ItemSpec);
-                    continue;
-                }
-
-                if (!existingCompressionFormatsByAssetItemSpec.TryGetValue(relatedAssetItemSpec, out var existingFormats))
-                {
-                    existingFormats = new();
-                    existingCompressionFormatsByAssetItemSpec.Add(relatedAssetItemSpec, existingFormats);
-                }
-
-                var assetTraitValue = asset.GetMetadata("AssetTraitValue");
-                string assetFormat;
-
-                if (string.Equals(assetTraitValue, GzipAssetTraitValue, StringComparison.OrdinalIgnoreCase))
-                {
-                    assetFormat = GzipFormatName;
-                }
-                else if (string.Equals(assetTraitValue, BrotliAssetTraitValue, StringComparison.OrdinalIgnoreCase))
-                {
-                    assetFormat = BrotliFormatName;
-                }
-                else
-                {
-                    Log.LogError(
-                        "The asset '{0}' has an unknown compression format '{1}'.",
-                        asset.ItemSpec,
-                        assetTraitValue);
-                    continue;
-                }
-
-                Log.LogMessage(
-                    "The asset '{0}' with related asset '{1}' was detected as already compressed with format '{2}'.",
-                    asset.ItemSpec,
-                    relatedAssetItemSpec,
-                    assetFormat);
-                existingFormats.Add(assetFormat);
-            }
-        }
+        var candidates = CandidateAssets.Select(StaticWebAsset.FromTaskItem).ToArray();
+        var explicitAssets = ExplicitAssets?.Select(StaticWebAsset.FromTaskItem).ToArray() ?? [];
+        var existingCompressionFormatsByAssetItemSpec = CollectCompressedAssets(candidates);
```

Comment:
Updated the task to use StaticWebAsset instead of the taskitem as staticwebasset guarantees that all items are valid and normalized anything that can be a problem.